### DISCUSSION TITLE
[char*] How to bind a function accept a char* argument?

### DISCUSSION THREAD

**Discussion by huangweiwu** [wjakob/nanobind#268] [Q&A] [answered]
The code below comes up with a compile error.
"Return (char *)": cannot convert argument 1 from "const char *" to "char *".

`void test_func(char* c) {}
NB_MODULE(my_ext, m) {
    m.def("test_func", test_func, "c"_a);
}`

**Comment by wjakob** [comment, marked as answer]
It's not supported. The nanobind Python bindings will call your function with a `const char *` from an internal buffer that you are not allowed to write to.

**Reply by huangweiwu** [reply]
Get it!
